frontend/src/routes/ApplicationPortal.jsx

```diff
@@ -85,38 +85,53 @@ const ApplicationPortal = () => {
     return <div>Error: {error.message}</div>;
   } else if (isFetching) {
     return <LoadingBall />;
+  } else if (admission?.detail === "Not found.") {
+    return <NotFoundPage />;
```

Comment:
hmm, I'm guessing you don't have access to the http status code here. That would be alot better than this if possible.